crates/ide_db/src/source_change.rs

```diff
@@ -17,27 +23,51 @@ impl SourceChange {
     /// Creates a new SourceChange with the given label
     /// from the edits.
     pub fn from_edits(
-        source_file_edits: Vec<SourceFileEdit>,
+        source_file_edits: SourceFileEdits,
         file_system_edits: Vec<FileSystemEdit>,
     ) -> Self {
         SourceChange { source_file_edits, file_system_edits, is_snippet: false }
     }
 }
 
-#[derive(Debug, Clone)]
-pub struct SourceFileEdit {
-    pub file_id: FileId,
-    pub edit: TextEdit,
+#[derive(Default, Debug, Clone)]
+pub struct SourceFileEdits {
+    pub edits: FxHashMap<FileId, TextEdit>,
+}
```

Comment:
Does this type pull its weight? Seems like we can just use `SourceChange` directly, no?

Reply:
Ye its actually not needed at all, thought `AssistBuilder` used it on its own but I didn't notice that it actually copied all of the `SourceChange` fields